src/main/java/lotto/model/money/Money.java

```diff
@@ -6,11 +6,13 @@
     public static final String AMOUNT_POSITIVE_MESSAGE = "금액은 0 이상이어야 합니다.";
     public static final String MONEY_DIVIDABLE_BY_UNIT_MESSAGE = "금액은 천원 단위여야 합니다.";
 
-    private final int amount;
+    private final int originalAmount;
+    private int currentAmount;
 
     private Money(int amount) {
         validate(amount);
-        this.amount = amount;
+        originalAmount = amount;
+        this.currentAmount = amount;
```

Comment:
```suggestion
        this.originalAmount = amount;
        this.currentAmount = amount;
```
클래스이름 Money만 봤을 때 originalAmount, currentAmount 두개를 갖고 있을 거라는 생각이 안드네요 🤔

Reply:
그렇네요. Money는 아예 불변객체로 관리하는게 맞는 것 같아 불변 객체로 변경하면서 amount 하나만 남겨두도록 변경하였습니다!